lib/srv/db/postgres/engine.go

```diff
@@ -418,8 +419,12 @@ func (e *Engine) receiveFromServer(serverConn *pgconn.PgConn, serverErrCh chan<-
 	copyReader, copyWriter := io.Pipe()
 	defer copyWriter.Close()
 
+	wg := sync.WaitGroup{}
+	wg.Add(1)
```

Comment:
Why not just close a channel instead? If we only need to wait for a single thing we don't really need a wait _group_ right?

Reply:
+1 - a channel send or close would do fine.

Reply:
Eh? You can use channels to send multiple messages, too. Both have practically identical performance in this use case, which would be absolutely drowned by any incidental I/O. In my mind wait groups are simpler than channels, too.

Reply:
I'm not worried about performance, just semantics. Wait groups include a counter and require extra cognitive overhead to ensure the right amount of `Add`s are performed compared to `Done`s, and that the add happens before launching the goroutine.

That said, not a hill I want to die on. You're welcome to take this feedback and do what you choose with it.

Reply:
Same as Zac.

Reply:
> That said, not a hill I want to die on. 
Ah, same here, really. I'm sorry if you got a different impression, I was genuinely confused what might be the issue. I guess performance, but I value your perspective about cognitive load. `sync.WaitGroup` can certainly be done wrong, too, so this is valid.

I have switched the implementation now (https://github.com/gravitational/teleport/pull/32650/commits/34d4806133d9ebc9c92d106ff45868f930996aac), but I did some extra research around the code base. Looks like both approaches are in use, which begs the question: should we standardize on one approach? Right now, you need to be prepared to recognize both patterns, which doesn't help with readability.

My personal favorite would still be `sync.WaitGroup`, simply because it is intended for _waiting_ and it has limited API. Channels, on the other hand, are complex beasts that can do _too much_. It feels like a hack to use them in this fashion, even if it is a very traditional hack ;-) 

Anyway, I've got a lot more to say on this, but I don't think this is the right time or place.